### ISSUE TITLE
Pivot Table Help Needed

### ISSUE THREAD

**Issue by whitebird58** [dfinke/ImportExcel#758]
I'm trying to have a pivot table that has the Pivot data showing the sum of a column and then another pivot column that is the sum of the same data but presented as percent to grand total.  I can add two pivot columns but am at a loss as to how to make one the percent to grant total of the same value.  Additionally, I want the column that is being pivoted also in the pivot filters but I cannot add it directly.  And I admit that I'm not super good just in Excel.  Someone else created the pivot I'm trying to recreate in code.  I used one of the pivot examples as a starter and I was able to get two pivot data columns but don't know how to get one that is calculated.  See below which is "working" but not yet the results I really want.

-------------------------
Import-Module ImportExcel

$outfile = "$env:TEMP\PivotTest.xlsx"
Remove-Item -path $outfile -ErrorAction SilentlyContinue

$ExcelParams = @{
    Path              = $outFile
    WorksheetName     = "Sales"
    TableName         = "Details"
    AutoFilter        = $true
    AutoSize          = $true
    BoldTopRow        = $true
    IncludePivotTable = $true
    PivotDataToColumn = $true
    PivotTableName    = 'Pivot'
    PivotRows         = "Product"
    PivotData         = @{"units"="sum";"cost"="sum"}
    PivotFilter       = @("Region", "Area", "Product")
    PivotTotals       = "both"
    Activate          = $true
    AutoNameRange     = $true
    Verbose           = $true
}

$data = @"
Region,Area,Product,Units,Cost
North,A1,Apple,100,.5
South,A2,Pear,120,1.5
East,A3,Grape,140,2.5
West,A4,Banana,160,3.5
North,A1,Pear,120,1.5
North,A1,Grape,140,2.5
"@ 

$data | ConvertFrom-Csv | Export-Excel @ExcelParams -Show
--------------------------

The above works but you get the warning that Product cannot be added to the Pivot.  And in the PivotData, what I really want rather than a sum of cost is the sum of units again but then displayed as a percentage to grant total.  Straining my brain so some hints would be appreciated.

**Comment by dfinke** [comment]
I'm not an expert in Pivot Filters. I believe you cannot have a PivotRow cannot also be a Filter.
Remove "Product" from PivotFilter. You can then drag and drop fields in the Excel "dialog" on the right. If you drag "Product" onto the Filters list box, it will add it and then remove it from the Rows list box.

**Comment by jhoneill** [comment]
OK. We have multiple things going on. 
1. A field (from the headings of the source data) can't be a filter and a row / column heading. 
2. You CAN have the same field in data twice. BUT we don't support that because we use a hash table to say Column = Function  (sum, count etc.)  ; so you can only have one instance of the same data 
3. As a work-round for 2 you can use the Add-PivotTable command with -Passthrough and add the column you want to double up. 
4. The EPPlus library we use doesn't provide access to the display-as values, and I can't see way to hack the XML to set  % of , rank etc.  It must be in the XML somewhere,  but I can't find it.

**Comment by whitebird58** [comment]
OK I do see where there are things that I cannot do for this particular requirement.  So my next shot was to create a template excel and open it and replace the data in the detail data worksheet and have the pivot refresh on open.  Going to give that a try.  :)

**Comment by pkarunkar** [comment]
Check issue #521 . 

For now I have working around using com object to automate pivot filters. 
Import excel gets 90 percent of the work done. Few limitations that epplus has, I overcome with com object. 

At the end automation is automation . Whatever gets the work done. :)

**Comment by whitebird58** [comment]
pkarunkar - that only works if you have Excel installed.  I'm doing this on a server so don't have that option.  But thanks for the info.

**Comment by whitebird58** [comment]
I have solved my issue.  I used Export-Excel to create a worksheet with the following parameters:

$ExcelParams = @{
    Path              = $outFile
    WorksheetName     = "Scan Details"
    TableName         = "Details"
    AutoFilter        = $true
    AutoSize          = $true
    BoldTopRow        = $true
    ClearSheet        = $true
}

Then I created a pivot on another sheet and made it exactly like I wanted it.  I then deleted the "Scan Details" worksheet and saved the file.  Basically it just has a pivot table with no data to reference.  Then I copy that template file to a file I want to put the data in, and import the data and pipe to the same Export-Excel @ExcelParams.  I did set the pivot to refresh on file open so that it automatically refreshes for the users.  The ClearSheet is likely unnecessary but I thought I might play with leaving the sheet there and replacing the data and it doesn't hurt anything.  So with that, I will close this issue.  Thanks!
